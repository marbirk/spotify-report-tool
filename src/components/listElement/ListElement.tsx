import React from 'react'

interface ListElementProps {
    children: React.ReactNode
}

const ListElement = ({ children, ...props }: ListElementProps) => (
    <li className="pr-2 pl-2" {...props}>
        {children}
    </li>
)

export default ListElement
