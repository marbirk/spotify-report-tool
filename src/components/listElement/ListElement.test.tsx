import React from 'react'
import { render } from '@testing-library/react'

import ListElement from './ListElement'

describe('ListElement', () => {
    test('renders correctly', () => {
        const { container, getByText } = render(
            <ListElement>List Element</ListElement>
        )
        expect(getByText('List Element')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})
