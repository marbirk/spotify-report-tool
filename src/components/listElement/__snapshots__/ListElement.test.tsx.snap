// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListElement renders correctly 1`] = `
<div>
  <li
    class="pr-2 pl-2"
  >
    List Element
  </li>
</div>
`;
